### PR TITLE
Loading info

### PR DESCRIPTION
**Pull request by ajosephides**
A feature that presents a message to a user between pressing submit on the form and results being rendered.
This feature keeps a user informed and comforted.

Known issue: the feature test does not test for the message disappearing after a response is returned. This is because there was no way (that I could find) to delay the response from the intercepted request to give a consistent confidence in the test.

### CODE REVIEW

File: src/components/RecipesList.js [hunk 1]
```diff
@@ -25,6 +27,7 @@ class RecipesList extends React.Component {
   };
 
   onHandleSubmit(event) {
+    this.setState({showDataLoading: true});
```
Comment (bengscott2) [inline src/components/RecipesList.js:30]: May want to separate this out into its own function that gets triggered. Seems like this handle submit method is handling quite a lot of logic now.

File: src/components/RecipesList.js [hunk 2]
```diff
@@ -33,7 +36,7 @@ class RecipesList extends React.Component {
         let recipes = data.map((recipe) => {
           return {id:recipe.id, title:recipe.title, image:recipe.image};
         });
-        this.setState({recipes: recipes})
+        this.setState({recipes: recipes, showDataLoading: false})
```
Comment (bengscott2) [inline src/components/RecipesList.js:39]: Again could be a function call that switches true/ false based on what the current state

File: src/components/RecipesList.js [hunk 3]
```diff
@@ -58,6 +61,11 @@ class RecipesList extends React.Component {
           })
           }
         </ul>
+
```
Comment (bengscott2) [inline src/components/RecipesList.js]: Just for my own curiosity are there supposed to be blank lines around this div in terms of styling?

File: src/feature-test/user_see_data_pending.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      expect(pendingDataMessage).toBe(true)
+      
+      await browser.close();
+
+    } catch (error){
+      console.log(error)
+    }
+    
+  });
+});
```
Comment (bengscott2) [inline src/feature-test/user_see_data_pending.test.js:38]: Should have a new line at the end of the file.

File: src/unit-test/DataLoading.test.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+it('returns null if its prop is false', () => {
+  const wrapper = shallow(<DataLoading showDataLoading={ false }/>);
+  expect(wrapper.text()).toBe('');
+});
+
+it('returns the correct message if its prop is false', () => {
+  const wrapper = shallow(<DataLoading showDataLoading={ true }/>);
+  expect(wrapper.text()).toBe('Please wait while we get the data from our partners.....');
+});
```
Comment (bengscott2) [inline src/unit-test/DataLoading.test.js:12]: Well done on these tests.

**Review by bengscott2 — COMMENTED** [review]
Looks good just a few comments. Nothing that needs to be changed, just some thoughts. I agree with the way you tested for the data loading message and I think it's a fair assumption to make that if it's unit tested to show null and featured tested to make sure it does render then we can be sure that it's working properly.

File: src/components/RecipesList.js (hunk 1 above)
Comment (ajosephides) [reply src/components/RecipesList.js:30]: This did cross my mind but I reasoned that it was one line in the `onHandleSubmit` that would just be replaced by another one line that would call a function. The benefit of another separate function would be that you could give a reviewer a clear idea about what was being done - but again I felt this one line is pretty clear.
Later on in the method I opted to have one call that `setState` as thought could be more efficient. These two factors combined were what led me with the other current approach.

File: src/components/RecipesList.js (hunk 2 above)
Comment (ajosephides) [reply src/components/RecipesList.js:39]: Discussed above.

File: src/unit-test/DataLoading.test.js (hunk 5 above)
Comment (ajosephides) [reply src/unit-test/DataLoading.test.js:12]: Thanks. That really means a lot.

File: src/components/RecipesList.js (hunk 3 above)
Comment (ajosephides) [reply src/components/RecipesList.js]: I left over from how when I was working as it helps me see what i am working on. Will remove.

File: src/feature-test/user_see_data_pending.test.js (hunk 4 above)
Comment (ajosephides) [reply src/feature-test/user_see_data_pending.test.js:38]: As above

File: src/components/RecipesList.js (hunk 1 above)
Comment (bengscott2) [reply src/components/RecipesList.js:30]: I agree with your argument and I think it's fine. I would argue that the point of breaking it out into a separate function is less about line count and more about expandability. If it was in a separate function then you could easily change something if you needed to. Totally fine the way it is for now.

File: src/components/RecipesList.js (hunk 3 above)
Comment (bengscott2) [reply src/components/RecipesList.js]: 👍

**Review by bengscott2 — APPROVED** [review]
👍

File: src/components/RecipesList.js (hunk 2 above)
Comment (l-palermo) [inline src/components/RecipesList.js:39]: I agree with Ben. If it is not too much trouble separating it, would be better having a separate function for it.

File: src/feature-test/user_see_data_pending.test.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import '../setupTest.js'
+import { jsxEmptyExpression } from '@babel/types';
+import { SSL_OP_EPHEMERAL_RSA } from 'constants';
```
Comment (l-palermo) [inline src/feature-test/user_see_data_pending.test.js]: What is this? A dependency? What does it do?

Comment (ajosephides) [reply src/feature-test/user_see_data_pending.test.js]: Good catch. No idea where this came from.

File: src/components/RecipesList.js (hunk 2 above)
Comment (ajosephides) [reply src/components/RecipesList.js:39]: Cool.
I agree and we have discussed offline with rest of team.
Conclusion is that we will go ahead with this merge and be knowingly introduce some technical debt as a deadline is approaching and this refactor can happen in due course.

**Review by l-palermo — APPROVED** [review]

**Pull request merged**